runners/direct-java/src/main/java/org/apache/beam/runners/direct/SplittableProcessElementsEvaluatorFactory.java

```diff
@@ -46,16 +50,32 @@
     implements TransformEvaluatorFactory {
   private final ParDoEvaluatorFactory<KeyedWorkItem<String, KV<InputT, RestrictionT>>, OutputT>
       delegateFactory;
-  private final EvaluationContext evaluationContext;
+  private final ScheduledExecutorService ses = Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder()
+                    .setThreadFactory(MoreExecutors.platformThreadFactory())
+                    .setNameFormat("direct-splittable-process-element-checkpoint-executor")
+                    .build());
 
   SplittableProcessElementsEvaluatorFactory(EvaluationContext evaluationContext) {
-    this.evaluationContext = evaluationContext;
     this.delegateFactory =
-        new ParDoEvaluatorFactory<>(
-            evaluationContext,
-            SplittableProcessElementsEvaluatorFactory
-                .<InputT, OutputT, RestrictionT>processFnRunnerFactory(),
-            ParDoEvaluatorFactory.basicDoFnCacheLoader());
+      new ParDoEvaluatorFactory<>(
+        evaluationContext,
+        SplittableProcessElementsEvaluatorFactory.
+          <InputT, OutputT, RestrictionT>processFnRunnerFactory(),
+          new CacheLoader<AppliedPTransform<?, ?, ?>, DoFnLifecycleManager>() {
+            @Override
+            public DoFnLifecycleManager load(final AppliedPTransform<?, ?, ?> application) {
+              if (!ProcessElements.class.isInstance(application.getTransform())) {
+                throw new IllegalArgumentException(
+                  "No know extraction of the fn from " + application);
+              }
+              final ProcessElements<InputT, OutputT, RestrictionT, TrackerT> transform =
+                (ProcessElements<InputT, OutputT, RestrictionT, TrackerT>)
+                  application.getTransform();
+              final DoFn fn = ProcessElements.class.cast(application.getTransform()).getFn();
```

Comment:
This seems duplicated; why not `transform.getFn()` (and probably inline it into the `transform.newProcessFn`

Reply:
simplified, thanks